app/src/main/java/ch/epfl/sweng/SDP/home/HomeActivity.java

```diff
@@ -71,6 +79,41 @@ protected void onCreate(Bundle savedInstanceState) {
         LocalDbHandlerForAccount localDb = new LocalDbHandlerForAccount(this, null, 1);
         localDb.retrieveAccount(Account.getInstance(this));
 
+        Database.getReference(format("users.%s.friends", Account.getInstance(this).getUserId()))
+                .addValueEventListener(new ValueEventListener() {
+            @Override
+            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
+                for (DataSnapshot child : dataSnapshot.getChildren()) {
+                    Integer stateValue = child.getValue(Integer.class);
+                    if (stateValue != null) {
+                        FriendsRequestState state = FriendsRequestState.fromInteger(stateValue);
+
+                        if (state != null && state == FriendsRequestState.RECEIVED) {
```

Comment:
Replace this by if (state == FriendsRequestState.RECEIVED) {

Reply:
fromInteger() returns null if the given number isn't 0, 1 or 2, thus the check